src/exchange/passthrough_test.go

```diff
@@ -0,0 +1,1134 @@
+	require.True(t, a.CompletedAmount.Equal(b.CompletedAmount))
+	require.True(t, a.Fee.Equal(b.Fee))
+	require.True(t, a.CreateDate.Equal(b.CreateDate))
+	require.True(t, a.CompleteDate.Equal(b.CompleteDate))
+	require.Equal(t, a.OrderID, b.OrderID)
+	require.True(t, a.Price.Equal(b.Price))
+	require.Equal(t, a.Status, b.Status)
+	require.Equal(t, a.Type, b.Type)
+
+	if a.Trigger == nil {
+		require.Nil(t, b.Trigger)
+	} else {
+		require.True(t, a.Trigger.Equal(*b.Trigger))
+	}
+
+	if a.CustomerID == nil {
+		require.Nil(t, b.CustomerID)
+	} else {
+		require.Equal(t, *a.CustomerID, *b.CustomerID)
+	}
+
+	require.Equal(t, a.Source, b.Source)
+}
+
+func TestCalculateRequestedAmount(t *testing.T) {
+	cases := []struct {
+		name string
+		in   int64
+		out  string
+	}{
+		{
+			name: "zero",
+			in:   0,
+			out:  "0",
+		},
+		{
+			name: "1 satoshis too small, truncated",
+			in:   1,
+			out:  "0",
+		},
+		{
+			name: "10 satoshis too small, truncated",
+			in:   10,
+			out:  "0",
+		},
+		{
+			name: "100 satoshis too small, truncated",
+			in:   100,
+			out:  "0",
+		},
+		{
+			name: "1000 satoshis smallest value that won't truncate",
+			in:   1000,
+			out:  "0.00001",
+		},
+		{
+			name: "10000 satoshis",
+			in:   10000,
+			out:  "0.0001",
+		},
+		{
+			name: "100000 satoshis",
+			in:   100000,
+			out:  "0.001",
+		},
+		{
+			name: "1000000 satoshis",
+			in:   1000000,
+			out:  "0.01",
+		},
+		{
+			name: "10000000 satoshis (1 BTC)",
+			in:   10000000,
+			out:  "0.1",
+		},
+		{
+			name: "100000000 satoshis (1 BTC)",
+			in:   100000000,
+			out:  "1",
+		},
+		{
+			name: "mixed with truncation",
+			in:   92045678111,
+			out:  "920.45678",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			amt := calculateRequestedAmount(tc.in)
+			require.Equal(t, tc.out, amt.String())
+		})
+	}
+}
+
+func TestCalculateSkyBought(t *testing.T) {
+	cases := []struct {
+		name string
+		in   string
+		out  uint64
+		err  error
+	}{
+		{
+			name: "negative amount",
+			in:   "-1",
+			err:  errCompletedAmountNegative,
+		},
+		{
+			name: "one",
+			in:   "1",
+			out:  1e6,
+		},
+		{
+			name: "fractional",
+			in:   "1.23456",
+			out:  1234560,
+		},
+		{
+			name: "fractional exceeding 6 decimal places",
+			in:   "1.23456789",
+			out:  1234567,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			completedAmt, err := decimal.NewFromString(tc.in)
+			require.NoError(t, err)
+
+			amt, err := calculateSkyBought(&c2cx.Order{
+				CompletedAmount: completedAmt,
+			})
+
+			if tc.err != nil {
+				require.Equal(t, tc.err, err)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tc.out, amt)
+		})
+	}
+}
+
+func TestCalculateBtcSpent(t *testing.T) {
+	cases := []struct {
+		name   string
+		bought string
+		price  string
+		out    int64
+	}{
+		{
+			name:   "negative",
+			bought: "-1",
+			price:  "0.002",
+			out:    -200000,
```

Comment:
Why do we allow it to be negative?

Reply:
It should never be negative, just checking that the calculation doesn't freak out if it happened to see a negative value though